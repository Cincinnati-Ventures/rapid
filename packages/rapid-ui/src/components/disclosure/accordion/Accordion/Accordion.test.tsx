import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import '@testing-library/jest-dom/extend-expect';
import {
	Accordion,
	AccordionItem,
	AccordionHeader,
	AccordionIcon,
	AccordionContent,
} from '../';

const accordionDefault = (
	<div className='container mx-auto p-4'>
		<Accordion>
			<AccordionItem>
				<AccordionHeader>Item 1</AccordionHeader>
				<AccordionContent>Content for item 1</AccordionContent>
			</AccordionItem>
			<AccordionItem>
				<AccordionHeader>Item 2</AccordionHeader>
				<AccordionContent>Content for item 2</AccordionContent>
			</AccordionItem>
			<AccordionItem>
				<AccordionHeader>Item 3</AccordionHeader>
				<AccordionContent>Content for item 3</AccordionContent>
			</AccordionItem>
		</Accordion>
	</div>
);

const accordionWithProps = (
	<div className='container mx-auto p-4'>
		<Accordion allowMultiple allowToggle defaultIndexes={[0, 2]}>
			<AccordionItem>
				<AccordionHeader>
					Item 1<AccordionIcon />
				</AccordionHeader>
				<AccordionContent>
					Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
					do eiusmod tempor incididunt ut labore et dolore magna
					aliqua. Ut enim ad minim veniam, quis nostrud exercitation
					ullamco laboris nisi ut aliquip ex ea commodo consequat.
				</AccordionContent>
			</AccordionItem>
			<AccordionItem>
				<AccordionHeader>
					Item 2<AccordionIcon />
				</AccordionHeader>
				<AccordionContent>
					Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
					do eiusmod tempor incididunt ut labore et dolore magna
					aliqua. Ut enim ad minim veniam, quis nostrud exercitation
					ullamco laboris nisi ut aliquip ex ea commodo consequat.
				</AccordionContent>
			</AccordionItem>
			<AccordionItem>
				<AccordionHeader>
					Item 3<AccordionIcon />
				</AccordionHeader>
				<AccordionContent>
					Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
					do eiusmod tempor incididunt ut labore et dolore magna
					aliqua. Ut enim ad minim veniam, quis nostrud exercitation
					ullamco laboris nisi ut aliquip ex ea commodo consequat.
				</AccordionContent>
			</AccordionItem>
		</Accordion>
	</div>
);

describe('Accordion', () => {
	it('should render default accordion correctly', () => {
		const { container } = render(accordionDefault);
		expect(container).toMatchSnapshot();
	});

	it('should render accordion with props correctly', () => {
		const { container } = render(accordionWithProps);
		expect(container).toMatchSnapshot();
	});

	it('should render AccordionItem components', () => {
		const { container } = render(accordionDefault);
		const accordionItems = container.querySelectorAll(
			'.rapid-accordion-item',
		);
		expect(accordionItems.length).toBe(3);
	});

	it('should render AccordionHeader components', () => {
		const { container } = render(accordionDefault);
		const accordionHeaders = container.querySelectorAll(
			'.rapid-accordion-header',
		);
		expect(accordionHeaders.length).toBe(3);
	});

	it('should render AccordionContent components', () => {
		const { container } = render(accordionDefault);
		const accordionContents = container.querySelectorAll(
			'.rapid-accordion-content',
		);
		expect(accordionContents.length).toBe(3);
	});
});
