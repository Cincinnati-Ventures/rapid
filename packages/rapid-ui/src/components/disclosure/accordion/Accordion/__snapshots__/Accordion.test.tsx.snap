// Vitest Snapshot v1

exports[`Accordion > should render accordion with props correctly 1`] = `
<div>
  <div
    class="container mx-auto p-4"
  >
    <div
      aria-multiselectable="true"
      class="flex flex-col mt-5 p-3 border-2 rounded-lg overflow-x-auto rapid-accordion"
      role="region"
    >
      <div
        class="border-b rapid-accordion-item"
      >
        <h2
          aria-controls="rapid-accordion-content-0"
          aria-expanded="true"
          class="cursor-pointer p-2 hover:bg-lightGrey rapid-accordion-header"
          id="rapid-accordion-header-0"
          role="button"
          tabindex="0"
        >
          <div
            class="flex justify-between items-center"
          >
            Item 1
            <svg
              class="transition-transform duration-300 rotate-180 transform rapid-accordion-icon"
              height="16px"
              pointer-events="box-none"
              role="presentation"
              viewBox="0 0 448 512"
              width="16px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 338.7 54.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                fill="black"
              />
            </svg>
          </div>
        </h2>
        <div
          style="overflow: hidden; height: NaNpx;"
        >
          <div
            aria-labelledby="rapid-accordion-header-0"
            class="p-2 text-gray-500 rapid-accordion-content"
            id="rapid-accordion-content-0"
            role="region"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </div>
        </div>
      </div>
      <div
        class="border-b rapid-accordion-item"
      >
        <h2
          aria-controls="rapid-accordion-content-1"
          aria-expanded="false"
          class="cursor-pointer p-2 hover:bg-lightGrey rapid-accordion-header"
          id="rapid-accordion-header-1"
          role="button"
          tabindex="0"
        >
          <div
            class="flex justify-between items-center"
          >
            Item 2
            <svg
              class="transition-transform duration-300 rapid-accordion-icon"
              height="16px"
              pointer-events="box-none"
              role="presentation"
              viewBox="0 0 448 512"
              width="16px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 338.7 54.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                fill="black"
              />
            </svg>
          </div>
        </h2>
        <div
          style="overflow: hidden; height: 0px;"
        >
          <div
            aria-labelledby="rapid-accordion-header-1"
            class="p-2 text-gray-500 rapid-accordion-content"
            id="rapid-accordion-content-1"
            role="region"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </div>
        </div>
      </div>
      <div
        class="border-b rapid-accordion-item"
      >
        <h2
          aria-controls="rapid-accordion-content-2"
          aria-expanded="true"
          class="cursor-pointer p-2 hover:bg-lightGrey rapid-accordion-header"
          id="rapid-accordion-header-2"
          role="button"
          tabindex="0"
        >
          <div
            class="flex justify-between items-center"
          >
            Item 3
            <svg
              class="transition-transform duration-300 rotate-180 transform rapid-accordion-icon"
              height="16px"
              pointer-events="box-none"
              role="presentation"
              viewBox="0 0 448 512"
              width="16px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M201.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 338.7 54.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                fill="black"
              />
            </svg>
          </div>
        </h2>
        <div
          style="overflow: hidden; height: NaNpx;"
        >
          <div
            aria-labelledby="rapid-accordion-header-2"
            class="p-2 text-gray-500 rapid-accordion-content"
            id="rapid-accordion-content-2"
            role="region"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordion > should render default accordion correctly 1`] = `
<div>
  <div
    class="container mx-auto p-4"
  >
    <div
      aria-multiselectable="false"
      class="flex flex-col mt-5 p-3 border-2 rounded-lg overflow-x-auto rapid-accordion"
      role="region"
    >
      <div
        class="border-b rapid-accordion-item"
      >
        <h2
          aria-controls="rapid-accordion-content-0"
          aria-expanded="false"
          class="cursor-pointer p-2 hover:bg-lightGrey rapid-accordion-header"
          id="rapid-accordion-header-0"
          role="button"
          tabindex="0"
        >
          <div
            class="flex justify-between items-center"
          >
            Item 1
          </div>
        </h2>
        <div
          style="overflow: hidden; height: 0px;"
        >
          <div
            aria-labelledby="rapid-accordion-header-0"
            class="p-2 text-gray-500 rapid-accordion-content"
            id="rapid-accordion-content-0"
            role="region"
          >
            Content for item 1
          </div>
        </div>
      </div>
      <div
        class="border-b rapid-accordion-item"
      >
        <h2
          aria-controls="rapid-accordion-content-1"
          aria-expanded="false"
          class="cursor-pointer p-2 hover:bg-lightGrey rapid-accordion-header"
          id="rapid-accordion-header-1"
          role="button"
          tabindex="0"
        >
          <div
            class="flex justify-between items-center"
          >
            Item 2
          </div>
        </h2>
        <div
          style="overflow: hidden; height: 0px;"
        >
          <div
            aria-labelledby="rapid-accordion-header-1"
            class="p-2 text-gray-500 rapid-accordion-content"
            id="rapid-accordion-content-1"
            role="region"
          >
            Content for item 2
          </div>
        </div>
      </div>
      <div
        class="border-b rapid-accordion-item"
      >
        <h2
          aria-controls="rapid-accordion-content-2"
          aria-expanded="false"
          class="cursor-pointer p-2 hover:bg-lightGrey rapid-accordion-header"
          id="rapid-accordion-header-2"
          role="button"
          tabindex="0"
        >
          <div
            class="flex justify-between items-center"
          >
            Item 3
          </div>
        </h2>
        <div
          style="overflow: hidden; height: 0px;"
        >
          <div
            aria-labelledby="rapid-accordion-header-2"
            class="p-2 text-gray-500 rapid-accordion-content"
            id="rapid-accordion-content-2"
            role="region"
          >
            Content for item 3
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
